MailAlert.php

```diff
@@ -171,8 +171,10 @@ public static function sendCustomerAlert($id_product, $id_product_attribute)
     {
         $link = new Link();
         $context = Context::getContext()->cloneContext();
```

Comment:
Here the context should be cloned, such that the change of the shop id below (`$context->shop->id = $id_shop;`) should not affect the global context.
As the [PHP doc](https://www.php.net/manual/en/language.oop5.cloning.php) says:
> When an object is cloned, PHP will perform a shallow copy of all of the object's properties. Any properties that are references to other variables will remain references.
`$context->shop` is a reference to a `Shop` object, thus the global context gets altered despite of the call to `cloneContext()`.

We could remove the line `$context = Context::getContext()->cloneContext();`. It's presence may lead the reader of the code to a misunderstanding

Reply:
The context switch was useless anyway. Now it works.